apps/web/app/(auth)/auth/login/page.tsx

```diff
@@ -9,14 +10,24 @@ export const metadata: Metadata = {
 };
 
 export default function SignInPage() {
+  const publicSignUpEnabled = env.SIGNUP_DISABLED !== "1";
```

Comment:
Would it maybe make sense to move the definition of the variables, e.g.
```const publicSignUpEnabled = env.SIGNUP_DISABLED !== "1";```
to the `constants.ts` file so we don't have to make these comparisons all over again?
But then we need to make sure that they are not imported variables are on the server.

what would you think about a `constants-server.ts` file with the import of the `server-only` package to prevent importing them at the client and moving all the server side environment variables/constants there?

Reply:
that's a nice idea!

Also, the "use server" will not work bec it only allows exporting async functions and not const vals. Remember facing this issue when we were trying to export an interface from a service file!

Update, I took a different approach and instead of having a separate `constants` file, I imported all those env vars into server components and passed them as props to child client components! 😎

Reply:
I meant the server-only package as described here:
https://nextjs.org/docs/app/building-your-application/rendering/composition-patterns#keeping-server-only-code-out-of-the-client-environment

Reply:
oh okay, my bad, done!